NumLib/DOF/LocalToGlobalIndexMap.h

```diff
@@ -174,6 +189,17 @@ class LocalToGlobalIndexMap final
         std::vector<MeshLib::Element*> const& elements,
         NumLib::MeshComponentMap&& mesh_component_map);
 
+    /// Private constructor used by internally created local-to-global index
+    /// maps. The mesh_component_map is passed as argument instead of being
+    /// created by the constructor.
+    /// \attention The passed mesh_component_map is in undefined state after
+    /// this construtor.
+    explicit LocalToGlobalIndexMap(
+        std::vector<std::unique_ptr<MeshLib::MeshSubsets>>&& mesh_subsets,
+        std::vector<std::size_t> const global_component_ids,
```

Comment:
In the ctor above the component id is `int`.
The vector is passed by value. Is that intended? If so, then IMO there should be a comment that it is done so deliberately.